### PR TITLE
chore(settings): fix error when adding network

### LINKED ISSUE #544: Block polling fails on adding a new network
## Expected Behavior
Add a network and it starts polling blocks

## Current Behavior
It starts polling at `localhost:9080/NetworkName/v1/get_all_nodes` using the network name instead of the url.

## Steps to Reproduce
1. Add network and watch at the error

## Your Environment
* nOS Client version used: dev branch
* Environment name and version
    * Node version:
    * NPM version:
    * Yarn version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by mhuggins**
## Description
This fixes the selection of newly added networks to prevent JS errors & failed network polling.

## Motivation and Context
The network selection should only occur once it's available in the redux store.  Selecting prior will result in an invalid state.

## How Has This Been Tested?
Adding a new network to ensure the new network is selected properly.

## Screenshots (if appropriate)
N/A

## Types of changes
- [x] Chore (tests, refactors, and fixes)
- [ ] New feature (adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [x] I have read the **[CONTRIBUTING](./CONTRIBUTING.md)** guidelines and confirm that my code follows the code style of this project.
- [ ] Tests for the changes have been added (for bug fixes/features)

#### Documentation
- [ ] Docs need to be added/updated (for bug fixes/features)

## Closing issues
Fixes #544

### CODE REVIEW

**Comment by DalderupMaurice** [comment]
On a side note - currently when you have the wallet open
and you're suddenly connected to another node which is a bad node, the wallet disappears and you get the FAILED TO LOAD message.
I think this shouldn't happen, but instead on the background try to reconnect to good nodes. Is this something also covered in this PR? It's hard to test :P

File: src/renderer/root/components/App/index.js [hunk 1]
```diff
@@ -18,14 +19,33 @@ const { LOADING, LOADED } = progressValues;
...
   withInitialCall(currencyActions),
   withInitialCall(currentNetworkActions),
   withInitialCall(getAllNetworks),
   withData(getAllNetworks, mapNetworksToProps),
-  withProgressChange(getAllNetworks, LOADED, (state, { networks }) => updateNetworks(networks)),
+  withProgressChange(getAllNetworks, LOADED, (state, props) => updateNetworks(props.networks)),
+
+  // Change to new network whenever a new one is added
+  // TODO: This logic should really go into the `NetworkSettings` container, but it's not working
+  //       there.  Both the prevProps and nextProps `progress` are evaluating to "LOADED",
```
[the remark this answers is not in the document]
Comment (mhuggins) [reply src/renderer/root/components/App/index.js]: There are four possible states: INITIAL, LOADING, LOADED, and FAILED.  LOADED simply means that the action completed successfully.  If the action is executing, it's in state LOADING, and if it's finished successfully, it's LOADED.

With regards to `getAllNetworks`, this function actually shares a redux store key with `addNetwork` (see networkActions.js).  So when that action progress is LOADED, that means either the networks have been loaded for the first time, or a new network has been added.  The extra logic below on line 41-47 check if a new network was added, and if it was, then we dispatch an action to switch to that new network.

**Comment by mhuggins** [comment]
> Is this something also covered in this PR?

No, I think #616 will cover that.

**Comment by jeroenptrs** [comment]
@DalderupMaurice @mhuggins will take this up tomorrow guys!

**Comment by jeroenptrs** [comment]
@mhuggins This is good to merge, everything seems to work 😄

**Pull request merged**